common.go

```diff
@@ -3,11 +3,93 @@ package jwt
 // The setting are used by TokenProvider and TokenValidator.
 type CommonTokenConfig struct {
-	TokenName   string `json:"token_name,omitempty" xml:"token_name" yaml:"token_name"`
-	TokenSecret string `json:"token_secret,omitempty" xml:"token_secret" yaml:"token_secret"`
+	TokenName     string `json:"token_name,omitempty" xml:"token_name" yaml:"token_name"`
+	TokenHSSecret string `json:"token_secret,omitempty" xml:"token_secret" yaml:"token_secret"`
+	TokenRS
 	TokenIssuer string `json:"token_issuer,omitempty" xml:"token_issuer" yaml:"token_issuer"`
 	TokenOrigin string `json:"token_origin,omitempty" xml:"token_origin" yaml:"token_issuer"`
 	// The expiration time of a token in seconds
 	TokenLifetime      int    `json:"token_lifetime,omitempty" xml:"token_lifetime" yaml:"token_lifetime"`
 	TokenSigningMethod string `json:"token_signing_method,omitempty" xml:"token_signing_method" yaml:"token_signing_method"`
 }
+
+// TokenRS holds data for RSA keys that can be used to sign and verify JWT tokens
+// TokenRSDirectory is a directory that is like:
+//
+// <kid>'s can only contain ascii letters/numbers and underscores. (otherwise they are not loaded)
+//
+// <dirname>
+//    +-- <kid_1>
+//          +-- private.key
+//    +-- <kid_2>
+//          +-- public.key
+//    +-- kid_3.key
+//    +-- kid_4.key
+//    +-- kid.5.key
+// The above directory will result in a TokenRSKeys that looks like:
+//
+// TokenRSKeys{
+//     "kid_1_private": "---- RSA PRIVATE KEY ---- ...",
+//     "kid_2_public": "---- RSA PUBLIC KEY ---- ...",
+//     "kid_3": "---- RSA PRIVATE KEY ---- ...",
+//     "kid_4": "---- RSA PUBLIC KEY ---- ...",
+//     // there is no "kid.5" becuase the "." is invalid.
+// }
+//
+// There only needs to be public keys loaded for verification. If you're using the Grantor method then
+// you need to load a PrivateKey so that keys can be signed.
+//
+// The TokenRS fields translate to the following config values:
+//
+// "token_rs_dir": "<path to dir>"
+// "token_rs_files": {"<kid>": "<path to file>", ...}
+// "token_rs_keys": {"<kid>": "<key PEM value>", ...}
+//
+// there are two special config values:
+//
+// "token_rs_file": "<path to file>"
+// "token_rs_key": "<key PEM value>"
+//
+// The above two variables map to a <kid> of "0", these are always evaluated first so they can be overwritten if
+// a <kid> of "0" is used explictly
+//
+// The TokenRS fields translate to the following enviornment variables:
+//
+// JWT_RS_DIR="<path to dir>"
+// JWT_RS_FILE_<kid>="<path to file>"
+// JWT_RS_KEY_<kid>="<key PEM value>"
+//
+// there are two special environment variables:
+//
+// JWT_RS_FILE="<path to file>"
+// JWT_RS_KEY="<key PEM value>"
+//
+// The above two variables map to a <kid> of "0", these are always evaluated first so they can be overwritten if
+// a <kid> of "0" is used explictly
+//
+// Enviroment variable KID's get lowercased. All other KID's are left untouched.
+type TokenRS struct {
```

Comment:
@njones , please rename this to `RsaTokenConfig`. Also, I think it is hard for people to understand RS/HS... abbreviations do not make it simple.

Reply:
@njones . perhaps `RsaSignMethodConfig` is better wording.

Reply:
In respect to https://github.com/golang/go/wiki/CodeReviewComments#initialisms how about `RSASignMethodConfig`? (I'd like to note that I used `RS` as the acronym because that's what the JWT specification uses as RS256, RS512, etc..., so I'd like to suggest `RSMethodConfig`, thoughts?)